### ISSUE TITLE
is there a way to ssh into a child cluster

### ISSUE THREAD

**Issue by jasine** [clusternet/clusternet#264]
#### What would you like to be added:
  
- way to ssh into child cluster
- way to exec into remote pods 
- way to print log through clusternet plugin

#### Why is this needed:

ssh into child cluster or exec into pods of child cluster is necessary in edge scenario, to do some maintenance work or check logs

**Comment by dixudx** [comment]
> way to ssh into child cluster

I don't get what does this mean. The kubelet nodes?

> way to exec into remote pods
way to print log through clusternet plugin

Yes you can. You can just directly use kubectl clusternet plugin to do all these operations, directly from parent cluster. You can refer to [this](https://github.com/clusternet/clusternet/blob/main/docs/tutorials/visiting-child-clusters-with-rbac.md#an-easy-way---kubectl-clusternet-plugin-recommended) for guidance.

```bash
$ kubectl clusternet --cluster-id=CLUSTER-ID --child-kubeconfig=./YOUR-CONFIG-kubeconfig exec -it -n my-ns my-pod bash
```

```bash
$ kubectl clusternet --cluster-id=CLUSTER-ID --child-kubeconfig=./YOUR-CONFIG-kubeconfig logs -n my-ns my-pod 
```

Please install `clusternet` plugin with newest version `v0.6.0`.

you can install Clusternet kubectl plugin with,

```
$ kubectl krew update
$ kubectl krew install clusternet
# check plugin version
$ kubectl clusternet version
```

or update existing Clusternet plugin to latest,
```
$ kubectl krew update
$ kubectl krew upgrade clusternet
# check plugin version
$ kubectl clusternet version
```

**Comment by jasine** [comment]
@dixudx sorry I did not use the last version of plugin, and log/exec works after upgrading to version 0.6.0

my first question is, we use npu/gpu in egde  scenario, and thus we have to upgrade drivers on kubelet node, so is there a way to ssh into node, do you have any suggesting

when it comes to upgrade, how can we upgrade clusternet-agent itself by helm chart

appreciate your answering

**Comment by dixudx** [comment]
> my first question is, we use npu/gpu in egde scenario, and thus we have to upgrade drivers on kubelet node, so is there a way to ssh into node, do you have any suggesting

Currently ssh into node is not supported. What Clusternet supports now are all Kubernetes native operations.

But this seems feasible if we add this feature into `clusternet-agent`. There are also some prerequisites. The nodes you want to ssh into should have port 22 open.

> when it comes to upgrade, how can we upgrade clusternet-agent itself by helm chart

Actually when a cluster is registered as member cluster, you can visit and modify all resources in the member cluster. That means you could reply on `kubectl clusternet` plugin to modify and upgrade `clusternet-agent` deployment to your desired version and configurations.

For helm chart, currently you can use `helm upgrade` per cluster manually.
Also you can construct a dedicate kubeconfig for the target cluster, then use `helm upgrade` directly towards parent cluster. You can follow [this guide](https://github.com/clusternet/clusternet/blob/main/docs/tutorials/visiting-child-clusters-with-rbac.md#a-hard-way---constructing-a-dedicated-kubeconfig) to learn how to make such a dedicate kubeconfig for child cluster.

> workloads can be promoted in karmada(https://github.com/karmada-io/karmada/blob/master/docs/userguide/promote-legacy-workload.md)

This seems to be another question. Currently this scenario is not supported yet. But this seems interesting. Maybe we should support this later in our plugin.

If this is supported, I think we may use this feature to help promote `clusternet-agent`, so that we could manage this component in the upper parent cluster.

**Comment by jasine** [comment]
@dixudx thank for you reply, looking forward to these features

**Comment by dixudx** [comment]
> when it comes to upgrade, how can we upgrade clusternet-agent itself by helm chart

@jasine Sorry for the late reply. Actually `clusternet-agent` itself can be managed as applications in Clusternet as well.

After we install `clusternet-agent` and get the `ManagedCluster` registered, we can create a `Subscription` to deploy `clusternet-agent` Deployment to those child clusters. Then we can modify this Deployment in parent cluster to upgrade `clusternet-agent` in child clusters.

**Comment by dixudx** [comment]
@jasine As the other question on ssh, we could rely on `port-forward` capability. As we know, all the child clusters can be accessed directly from parent cluster. If we can deploy a pod dedicated for `ssh` in every child clusters, then we can redirect our local ssh requests to remote targets through that `ssh` pod.

**Comment by jasine** [comment]
Is `port-forward` already available? @dixudx

**Comment by dixudx** [comment]
> Is `port-forward` already available? @dixudx

It is easy to implement `port-forward` to pods/services in child clusters, which is working in progress. Hopefully this will be released in `kubectl-clusternet` v0.7.0.

**Comment by dixudx** [comment]
@jasine The `port-forward` feature has been available since `kubectl-clusternet` v0.7.0. You can have a try. Sorry for the late reply.

**Comment by jasine** [comment]
@dixudx thanks
